builder/vsphere/common/step_remote_upload.go

```diff
@@ -86,20 +92,21 @@ func (s *StepRemoteUpload) Cleanup(state multistep.StateBag) {
 	ui := state.Get("ui").(packer.Ui)
 	d := state.Get("driver").(*driver.VCenterDriver)
 
-	if UploadedCDPath, ok := state.GetOk("cd_path"); ok {
-		ui.Say("Deleting cd_files image from remote datastore ...")
-
-		ds, err := d.FindDatastore(s.Datastore, s.Host)
-		if err != nil {
-			state.Put("error", err)
-			return
-		}
-
-		err = ds.Delete(UploadedCDPath.(string))
-		if err != nil {
-			state.Put("error", err)
-			return
+	if s.UploadedCustomCD {
+		if UploadedCDPath, ok := state.GetOk("cd_path"); ok {
+			ui.Say("Deleting cd_files image from remote datastore ...")
+
+			ds, err := d.FindDatastore(s.Datastore, s.Host)
+			if err != nil {
+				log.Printf("Error finding datastore to delete custom CD; please delete manually: %s", err)
+				return
+			}
+
+			err = ds.Delete(UploadedCDPath.(string))
+			if err != nil {
+				log.Printf("Error deleting custom CD from remote datastore; please delete manually: %s", err)
+				return
+			}
```

Comment:
Nit: We can use this second if as a guard clause a return immediately if this is nothing to delete. 
```go
		if UploadedCDPath, ok := state.GetOk("cd_path"); !ok {
		    return
		}
		ui.Say("Deleting cd_files image from remote datastore ...")

		ds, err := d.FindDatastore(s.Datastore, s.Host)
		if err != nil {
			log.Printf("Error finding datastore to delete custom CD; please delete manually: %s", err)
			return
		}

		err = ds.Delete(UploadedCDPath.(string))
		if err != nil {
			log.Printf("Error deleting custom CD from remote datastore; please delete manually: %s", err)
			return
		}
```

Reply:
If we do it the way you've written UploadedCDPath is scoped in such a way that we can't use it later. We'd have to change it to 

```	
UploadedCDPath, ok := state.GetOk("cd_path")
	if !ok {
		return
	}
```

Reply:
Ah yes, my bad. Oversight on that line.